modules/local/dvpolish/createFinalAsm.nf

```diff
@@ -0,0 +1,77 @@
+process DVPOLISH_CREATE_FINALASM {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda 'bioconda::seqkit=2.8.2'
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/seqkit:2.8.2--h9ee0642_0' :
+        'nf-core/seqkit:2.8.2--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(unpol_fasta), path(unpol_merqury_csv)  // meta map, unpolished assembly, corresponding merqury qv file
+    tuple val(meta2), path(pol_fasta), path(pol_merqury_csv)     // meta map, polished assembly, corresponding merqury qv file
+
+    output:
+    tuple val(meta), path('*.fasta.gz')  , emit: fasta_gz
+    path  "versions.yml"                 , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args       = task.ext.args ?: ''
+    def prefix     = task.ext.prefix ?: "${meta.id}"
+    def unpol_name = unpol_fasta.getBaseName()
+    def unpol_ext  = unpol_fasta.getExtension()
+    def pol_name   = pol_fasta.getBaseName()
+    def pol_ext    = pol_fasta.getExtension()
+
+    """
+    nl_unpol_ASM=\$(wc -l < ${unpol_merqury_csv})
+    nl_pol_ASM=\$(wc -l < ${pol_merqury_csv})
+
+    if [[ \${nl_unpol_ASM} -ne \${nl_pol_ASM} ]]
+    then 
+        >&2 echo "[ERROR] DVPOLISH_CREATE_FINALASM: merqury files have different lines: ${unpol_merqury_csv}: \${nl_unpol_ASM} != ${pol_merqury_csv}: \${nl_pol_ASM}"
+        exit 1
+    fi
+    
+    #split unpolished assembly by sequence ID
+    seqkit split -i -O unpolished_asm ${unpol_fasta}
+
+    #split polished assembly by sequence ID
+    seqkit split -i -O polished_asm ${pol_fasta}
+
+    l=1 
+    while [[ \$l -le \${nl_pol_ASM} ]]
+    do 
+        IFS='\t' read -r -a l_uasm <<< "\$(sed -n \${l}p ${unpol_merqury_csv})"
```

Comment:
It would be helpful if you could annotate these more cryptic lines with what they're achieving.